src/DFGNode.cpp

```diff
@@ -295,19 +391,47 @@ string DFGNode::getOpcodeName() {
       Function *func = ((CallInst*)m_inst)->getCalledFunction();
       if (func) {
         string newName = func->getName().str();
-	string removingPattern = "llvm.vector.";
-	int pos = newName.find(removingPattern);
-	if (pos == -1)
-	  pos = newName.find("llvm.");
-	newName.erase(pos, removingPattern.length());
+        string removingPattern = "llvm.vector.";
+        int pos = newName.find(removingPattern);
+        if (pos == -1)
+        pos = newName.find("llvm.");
+	      newName.erase(pos, removingPattern.length());
         string delimiter = ".v";
         newName = newName.substr(0, newName.find(delimiter));
-	replace(newName.begin(), newName.end(), '.', '_');
-	return newName;
+	      replace(newName.begin(), newName.end(), '.', '_');
+        return newName;
       }
       else
         return "indirect call";
     }
+    // for the special operations
+    else if (m_opcodeName.compare("call") == 0) {
+      Function *func = ((CallInst*)m_inst)->getCalledFunction();
+      if (func) {
+        string newName = func->getName().str();
+        // cout << "[DEBUG] demangle name: " << demangle(newName) << endl;
+        if (demangle(newName) == "lut(float)" || demangle(newName) == "lut(int)") {
```

Comment:
Can you briefly explain what `demangle` can do? Seems like it return the func name with argument data type?

Reply:
u are right.